pkg/ceph/osd/agent_test.go

```diff
@@ -521,25 +318,47 @@ func TestGetPartitionPerfScheme(t *testing.T) {
 				currOsdID++
 				return fmt.Sprintf(`{"osdid": %d}`, currOsdID), nil
 			}
-			return "", fmt.Errorf("unexpected mon_command '%v'", args)
+			return "", fmt.Errorf("unexpected command '%v'", args)
+		},
+		MockExecuteCommandWithOutput: func(debug bool, actionName string, command string, args ...string) (string, error) {
+			logger.Infof("Command: %s %v", command, args)
+			if command == "lsblk" {
+				if args[0] == "/dev/sda" {
+					return `NAME="sda" SIZE="107374182400" TYPE="disk" PKNAME=""`, nil
+				}
+				if args[0] == "/dev/sdb" {
+					return `NAME="sdb" SIZE="107374182400" TYPE="disk" PKNAME=""`, nil
+				}
+				if args[0] == "/dev/sdc" {
+					return `NAME="sdc" SIZE="44158681088" TYPE="disk" PKNAME=""`, nil
+				}
+			}
+			if command == "blkid" {
+				return "", nil
+			}
+			if command == "df" {
+				return "", nil
+			}
+			return "", fmt.Errorf("unexpected command %s %v", command, args)
 		},
 	}
 	context.Executor = executor
 
+	devices, err := getAvailableDevices(context, "sda,sdb", "sdc", false)
+	assert.Nil(t, err)
 	scheme, err := a.getPartitionPerfScheme(context, devices)
 	assert.Nil(t, err)
 	assert.Equal(t, 2, len(scheme.Entries))
 
-	// verify the metadata entries, they should be on sdc and there should be 4 of them (2 per OSD)
-	assert.NotNil(t, scheme.Metadata)
+	// verify the metadata entries, they should be on sdc and there should be 2 of them (2 per OSD)
+	require.NotNil(t, scheme.Metadata)
 	assert.Equal(t, "sdc", scheme.Metadata.Device)
 	assert.Equal(t, 4, len(scheme.Metadata.Partitions))
 
-	// verify the first entry in the performance partition scheme.  note that the block device will either be sda or
-	// sdb because ordering of map traversal in golang isn't guaranteed.  Ensure that the first is either sda or sdb
-	// and that the second is the other one.
+	// verify the first entry in the performance partition scheme.
```

Comment:
it'd be nice to restore this comment explaining why the ordering of the devices will be random.  also, we can remove the duplicated comment a few lines below on 361

Reply:
agreed, will add them back.